app/controllers/categories_controller.rb

```diff
@@ -0,0 +1,12 @@
+class CategoriesController < ApplicationController
+
+  def index
+    @categories = Category.all
+  end
+
+  def show
+    @category = Category.find(params[:id])
+    @products = @category.products
+  end
+
```

Comment:
Remove the unnecessary blank line preceding the index action and following the show action.